internal/metrics/metrics.go

```diff
@@ -0,0 +1,73 @@
+package metrics
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+var (
+	nodeRestartReason = prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Name: "eo_elasticsearch_cr_node_restart_reason",
+			Help: "Node restarted due to Cert Restart or Rolling Restart (1 = Cert Restart, 2 = Rolling Restart, 3 = Scheduled).",
+		})
+
+	rollingRestartsCounter = prometheus.NewCounter(
+		prometheus.CounterOpts{
+			Name: "eo_elasticsearch_cr_rolling_restart_counter",
+			Help: "Number of times the nodes restarted due to Rolling Restart.",
+		})
+
+	esClusterManagementState = prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Name: "eo_elasticsearch_cr_cluster_state",
+			Help: "Elasticsearch cluster management state is Managed or Unmanaged (1 = Managed, 2 = Unmanaged).",
+		})
+
+	metricList = []prometheus.Collector{
+		nodeRestartReason,
+		rollingRestartsCounter,
+		esClusterManagementState,
+	}
+)
+
+//This function registers the custom metrics to the kubernetes controller-runtime default metrics.
+func RegisterCustomMetrics() {
+	for _, metric := range metricList {
+		metrics.Registry.MustRegister(metric)
+	}
+
+	//Set initial value for metrics as "0".
+	nodeRestartReason.Set(0)
+	esClusterManagementState.Set(0)
+}
+
+//Sets the metric value to "1" when the node restarts due to cert redeploy.
+func UpdateMetricNodeRestartDueToCert() {
+	nodeRestartReason.Set(1)
+}
+
+//Sets the metric value to "2" when the node restarts due to rolling update.
+func UpdateMetricNodeRestartDueToRolling() {
+	nodeRestartReason.Set(2)
+}
+
+//Sets the metric value to "3" when the node restarts due to cert redeploy or rolling update is scheduled for.
+func UpdateMetricNodeRestartScheduled() {
+	nodeRestartReason.Set(3)
+}
+
+//Increment the metric value when the node restarts due to rolling update.
+func IncrementMetricNodeRestartRolling() {
+	rollingRestartsCounter.Inc()
+}
+
+//Sets the metric value to "1" when the ES Cluster Management State is Managed.
+func UpdateMetricESClusterManagementStateManaged() {
+	esClusterManagementState.Set(1)
```

Comment:
- Think of this case `esClusterManagementState.WithLabels(state = Managed).Set(10)`?
- Consider renaming to `SetEsClusterManagementStateManaged(int)`